### PR TITLE
fix: owner name for external files

### PR DESCRIPTION
**Pull request by yoavrozner**

### CODE REVIEW

File: src/views/File.vue [hunk 1]
```diff
@@ -74,16 +91,17 @@ export default {
     },
     onBreadcrumbClick(folder) {
       if (!folder) {
-        this.isFolderOwner()
-          ? this.$router.push("/my-drive")
-          : this.$router.push("/shared-with-me");
+        if (this.currentFile.appID === "drive") {
+          this.isFileOwner()
+            ? this.$router.push("/my-drive")
```
Comment (Shahar-Y) [inline src/views/File.vue:96]: What about a case of my file inside of a shared folder owned by someone else?

**Review by Shahar-Y — CHANGES_REQUESTED** [review]

File: src/views/File.vue [hunk 2]
```diff
@@ -42,50 +46,68 @@ export default {
     async getBreadcrumbs(folder) {
       if (!folder) return;
 
+      const hierarchy = folder.id
+        ? await filesApi.getFolderHierarchy(folder.id)
+        : [];
+      const getExternalNetworkFirstBreadcrumb = (appID) => {
+        const externalNetworkDest = this.externalNetworkDests.find(
```
Comment (nettashviro) [inline src/views/File.vue:53]: please use networkDest util (getNetworkItemByAppId), I didn't know "find" function, so you can change the "filter" function to "find" :)

File: src/views/File.vue [hunk 3]
```diff
@@ -42,50 +46,68 @@ export default {
     async getBreadcrumbs(folder) {
```
Comment (nettashviro) [inline src/views/File.vue:46]: very nice! for clear code and for easy reading, I'll separate the functions inside this function (getExternalNetworkFirstBreadcrumb, getDriveFirstBreadcrumb )

File: src/views/Folders.vue [hunk 4]
```diff
@@ -41,18 +41,34 @@ export default {
     async getBreadcrumbs(folder) {
       if (!folder) return;
 
+      const hierarchy = await filesApi.getFolderHierarchy(folder.id);
+
+      const getExternalNetworkFirstBreadcrumb = (appID) => {
```
Comment (nettashviro) [inline src/views/Folders.vue:46]: same note as before

**Review by nettashviro — CHANGES_REQUESTED** [review]

**Review by nettashviro — APPROVED** [review]

**Review by nettashviro — APPROVED** [review]

**Review by Shahar-Y — APPROVED** [review]

**Pull request merged**